packages/@aws-cdk/aws-cognito/lib/user-pool-idps/google.ts

```diff
@@ -37,13 +45,18 @@ export class UserPoolIdentityProviderGoogle extends UserPoolIdentityProviderBase
 
     const scopes = props.scopes ?? ['profile'];
 
+    //at least one of the properties must be configured
+    if ((!props.clientSecret || !props.clientSecret.trim()) && !props.clientSecretValue) {
+      throw new Error('Client Secret or Client Secret Value must be configured.');
+    }
```

Comment:
```suggestion
    if ((!props.clientSecret && !props.clientSecretValue) ||
      (props.clientSecret && props.clientSecretValue)) {
      throw new Error('Exactly one of "clientSecret" or "clientSecretValue" must be configured.');
    }
```

Should we just combine this into a single error message? I also don't think we
should do the `trim()` as part of the condition. If someone provided something
like `' '` I would just assume that they meant to.

Reply:
Done